src/main/java/frc/trigon/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    public boolean atTargetRightState() {
+        return Math.abs(toMeters(rightMotor.getSignal(TalonFXSignal.POSITION)) - targetState.positionMeters) < ClimberConstants.CLIMBER_TOLERANCE_METERS;
+    }
+
+    public boolean atTargetLeftState() {
+        return Math.abs(toMeters(leftMotor.getSignal(TalonFXSignal.POSITION)) - targetState.positionMeters) < ClimberConstants.CLIMBER_TOLERANCE_METERS;
+    }
+
+    void driveRightMotor(Measure<Voltage> voltageMeasure) {
+        rightMotor.setControl(voltageRequest.withOutput(voltageMeasure.in(Units.Volts)));
+    }
+
+    void driveLeftMotor(Measure<Voltage> voltageMeasure) {
+        leftMotor.setControl(voltageRequest.withOutput(voltageMeasure.in(Units.Volts)));
+    }
+
+    void setTargetState(ClimberConstants.ClimberState targetState) {
+        this.targetState = targetState;
+        setTargetPosition(targetState.positionMeters, targetState.positionMeters, targetState.affectedByRobotWeight);
+    }
+
+    void setTargetPosition(double targetRightPositionMeters, double targetLeftPositionMeters, boolean affectedByRobotWeight) {
+        rightMotor.setControl(determineRequest(affectedByRobotWeight)
+                .withPosition(targetRightPositionMeters)
+                .withFeedForward(calculateStayingInPlaceFeedback(rightMotor.getSignal(TalonFXSignal.POSITION), affectedByRobotWeight)));
+        leftMotor.setControl(determineRequest(affectedByRobotWeight)
+                .withPosition(targetLeftPositionMeters)
+                .withFeedForward(calculateStayingInPlaceFeedback(leftMotor.getSignal(TalonFXSignal.POSITION), affectedByRobotWeight)));
+    }
+
+    private DynamicMotionMagicVoltage determineRequest(boolean affectedByRobotWeight) {
+        return affectedByRobotWeight ? onChainPositionRequest : groundedPositionRequest;
+    }
+
+    private double calculateStayingInPlaceFeedback(double positionRotations, boolean affectedByRobotWeight) {
+        return affectedByRobotWeight ?
+                calculateParabola(positionRotations, ClimberConstants.ON_CHAIN_A, ClimberConstants.ON_CHAIN_B, ClimberConstants.ON_CHAIN_C)
+                : calculateParabola(positionRotations, ClimberConstants.GROUNDED_A, ClimberConstants.GROUNDED_B, ClimberConstants.GROUNDED_C);
+    }
+
+    private double calculateParabola(double x, double a, double b, double c) {
+        return a * Math.pow(x, 2) + b * x + c;
+    }
+
+    private void updateMechanisms() {
+        ClimberConstants.RIGHT_MECHANISM.update(targetState, rightMotor.getSignal(TalonFXSignal.POSITION), rightMotor.getSignal(TalonFXSignal.CLOSED_LOOP_REFERENCE));
+        ClimberConstants.LEFT_MECHANISM.update(targetState, leftMotor.getSignal(TalonFXSignal.POSITION), leftMotor.getSignal(TalonFXSignal.CLOSED_LOOP_REFERENCE));
+    }
+
+    double toMeters(double rotations) {
+        return Conversions.rotationsToDistance(rotations, ClimberConstants.DRUM_DIAMETER_METERS);
+    }
```

Comment:
if it's not private it should be above. This should only be in the visualization.